### ISSUE TITLE
error: Cannot determine type of a protected property

### ISSUE THREAD

**Issue by netokey** [python/mypy#9601]
**Bug Report**

Consider a simple class like below:

```python
class A:

    def __init__(self, a: int) -> None:
        self.num_a = a
        self.is_checked = False

    @property
    def num_a(self) -> int:
        return self._num

    @num_a.setter
    def num_a(self, a: int) -> None:
        self.is_checked = False
        self._num = a # error: Cannot determine type of '_num' 
```

**Expected Behavior**

recognize self._num with type 'int'

**Actual Behavior**

if change the order of the codes of 'num_a' setter like below:
```python
@num_a.setter
def num_a(self, a: int) -> None:
    self._num = a
    self.is_checked = False    
```
No errors raised.

**Your Environment**

- Mypy version used: 0.790
- Mypy command-line flags: no flag
- Mypy configuration options from `mypy.ini` (and other config files): default
- Python version used: 3.6.8
- Operating system and version: Windows 10

**Comment by AlexWaygood** [comment]
This still causes mypy to [emit an error](https://mypy-play.net/?mypy=latest&python=3.10&flags=warn-unreachable&gist=157d5d905fd4e6025cd16c0d54732256) on mypy 9.941. However, the error is now in the getter rather than the setter, which I think makes more sense:

```python
class A:
    def __init__(self, a: int) -> None:
        self.num_a = a
        self.is_checked = False

    @property
    def num_a(self) -> int:
        return self._num  # error: Cannot determine type of "_num"

    @num_a.setter
    def num_a(self, a: int) -> None:
        self.is_checked = False
        self._num = a
```

The error is also _very_ easily worked around by simply providing a type annotation for the `_num` attribute:

```python
class A:
    _num: int
    
    def __init__(self, a: int) -> None:
        self.num_a = a
        self.is_checked = False

    @property
    def num_a(self) -> int:
        return self._num

    @num_a.setter
    def num_a(self, a: int) -> None:
        self.is_checked = False
        self._num = a
```

So I'm going to go ahead and close this issue.
